### DISCUSSION TITLE
Smoother Transitions on Conditional UI

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7511] [Show and tell]
In this example, I implemented using an Embed, transition animations when a Container appear and when the Form Steps are changed from on to another.

https://user-images.githubusercontent.com/80512170/187433082-89ccac08-e214-4453-a163-0cb22d9c18de.mov

**Comment by ryzam** [comment]
Hi @poirazis , which components you are using to display a tab?

**Reply by poirazis** [reply]
Hey @ryzam 
The "tabs" are actually Buttons and the "tabContents" are Form Steps.
I use the buttons to navigate between the steps, and conditional UI to make it look li a Tab Control!


Hope this helps!

**Reply by ryzam** [reply]
If possible to share step by step, i think because its involve in custom css and conditional UI to achieve right?

**Reply by poirazis** [reply]
yes, let me create a small app with the implementation and post it here so you can play around with it. 
I will revert with a sample app.

**Reply by poirazis** [reply]
Hey Ryzam, 
you can find the sample app here : 
[TabsAndTransitions-export-1661959085587.txt](https://github.com/Budibase/budibase/files/9462750/TabsAndTransitions-export-1661959085587.txt)

A preview here :
https://user-images.githubusercontent.com/80512170/187715775-8b0a3e77-b3f7-4efa-a182-de2232a23630.mov

Please note that i use the names of the components to target them in the CSS i inject with the Embed and not alter the Spectrum classes. 
<img width="291" alt="Screenshot 2022-08-31 at 6 18 51 PM" src="https://user-images.githubusercontent.com/80512170/187716082-48ba8732-de7a-4ef4-854c-173afa1ed0ad.png">

Hope this helps

Have a blessed day.

**Reply by ryzam** [reply]
@poirazis  thank you so much .

**Comment by melohagan** [comment]
This is really slick 😎

**Comment by kasgithub8** [comment]
Hey @poirazis,

about the preview video, are you using a number to search through a table? 
If yes, how did you do that?

**Comment by poirazis** [comment]
Hey @kasgithub8 , I explain that process here https://github.com/Budibase/budibase/discussions/6294#discussioncomment-3518250

Hope this helps!

**Reply by kasgithub8** [reply]
> Hey @kasgithub8 , I explain that process here [#6294 (comment)](https://github.com/Budibase/budibase/discussions/6294#discussioncomment-3518250)
> 
> Hope this helps!

Im using internal tables unfortunately.
Do you have a tip on that?

**Comment by poirazis** [comment]
Sure! There's @melohagan original solution here https://github.com/Budibase/budibase/discussions/6294#discussioncomment-2938345 for internal tables!

**Reply by kasgithub8** [reply]
@poirazis I can't get behind it. 😭

**Reply by poirazis** [reply]
@kasgithub8 let me know of the basic scenario of what you need to achieve and i will try and make a sample app for you

**Reply by kasgithub8** [reply]
> @kasgithub8 let me know of the basic scenario of what you need to achieve and i will try and make a sample app for you

I basically really dislike the dynamic filter because you have to click so many times just to filter, I really wanted to find out how to implement a text field/number field (f.e. Auto ID) to filter that table with.

**Reply by poirazis** [reply]
have you checked out Static Filtering from the documentation ? https://docs.budibase.com/docs/searchfilter-data

**Reply by kasgithub8** [reply]
But the filter is pre-entered, no?

**Reply by poirazis** [reply]
Yes, the filter is pre-defined and the value the filter is looking for is bound to an input field.

Please see this sample app below

[FilteringSample-export-1662550345400.txt](https://github.com/Budibase/budibase/files/9505757/FilteringSample-export-1662550345400.txt)

**Reply by kasgithub8** [reply]
Oh my god, that is so beautiful. You're a goat.
